src/bin/driver.rs

```diff
@@ -549,10 +564,40 @@ mod app {
         c.local.driver_dac[channel as usize].set(current).unwrap()
     }
 
-    #[task(priority = 1, shared=[interlock_handle])]
+    #[task(priority = 1, shared=[interlock_handle, laser_interlock_pin])]
     fn trip_interlock(mut c: trip_interlock::Context) {
         c.shared.interlock_handle.lock(|handle| *handle = None);
-        log::error!("interlock tripped");
+        log::error!("thermostat interlock tripped");
+        c.shared.laser_interlock_pin.lock(|pin| pin.set_low());
+    }
+
+    #[task(priority = 1, shared=[telemetry, settings, laser_interlock_pin], local=[internal_adc])]
+    fn monitor_output(mut c: monitor_output::Context) {
+        let limits = c.shared.settings.lock(|settings| {
+            [
+                settings.low_noise.interlock_current,
+                settings.high_power.interlock_current,
+                settings.low_noise.interlock_voltage,
+                settings.high_power.interlock_voltage,
+            ]
+        });
+        let measurements = c.local.internal_adc.read_all();
+        for (limit, measurement) in limits.iter().zip(measurements.iter()) {
+            if (measurement.0 > *limit)
+                && c.shared
+                    .laser_interlock_pin
```

Comment:
This is wrong. You are reading the pin state in each iteration. And suppressing the print on multiple limit faults.
Read it once.